hibernate-core/src/main/java/org/hibernate/boot/model/internal/AnnotationBinder.java

```diff
@@ -693,14 +694,19 @@ public static Map<XClass, InheritanceState> buildInheritanceStates(
 					getSuperclassInheritanceState( clazz, inheritanceStatePerClass );
 			final InheritanceState state =
 					new InheritanceState( clazz, inheritanceStatePerClass, buildingContext );
+			final AnnotatedClassType classType = buildingContext.getMetadataCollector().getClassType( clazz );
+			if ( classType == EMBEDDABLE ) {
+				final String className = clazz.getName();
+				buildingContext.getMetadataCollector().addImport( unqualify( className ), className );
```

Comment:
Might be worth doing this conditionally?

```suggestion
				if ( !annotatedClass.isAnnotationPresent( Imported.class ) ) {
					buildingContext.getMetadataCollector().addImport( unqualify( className ), className );
				}
```